components/script/dom/xmlhttprequest.rs

```diff
@@ -1077,18 +1073,23 @@ impl XMLHttpRequest {
                 headers
                     .as_ref()
                     .map(|h| *self.response_headers.borrow_mut() = h.clone());
-
+                let len = headers.and_then(|h| h.typed_get::<ContentLength>());
+                let mut response = self.response.borrow_mut();
+                response.clear();
+                if let Some(len) = len {
+                    response.reserve(len.0 as usize);
```

Comment:
This should probably have a cap or at least be fallible, otherwise I can make a server auto-DOS Servo by sending a large content length.